## 主力出货条件统计分析

#### 获取1000支股票数据

In [1]:
import numpy as np
import pandas as pd
import yfinance as yf

In [2]:
df = pd.read_csv('./data/all_data_SS_2y.csv')
df = df.drop(df.columns[0], axis=1)
df

Date   Open   High    Low  Close  Adj Close    Volume       code
0       2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814  603123.SS
1       2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265  603123.SS
2       2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283  603123.SS
3       2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291  603123.SS
4       2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935  603123.SS
...            ...    ...    ...    ...    ...        ...       ...        ...
399969  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021  600108.SS
399970  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100  600108.SS
399971  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251  600108.SS
399972  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300  600108.SS
399973  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420  600108.SS

[399974 rows x 8 columns]

In [3]:
# 过滤掉code列中以'300'、'688'、‘8’开头的股票
df = df[~df['code'].str.startswith('300')]
df = df[~df['code'].str.startswith('688')]
df = df[~df['code'].str.startswith('8')]
df

Date   Open   High    Low  Close  Adj Close    Volume       code
0       2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814  603123.SS
1       2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265  603123.SS
2       2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283  603123.SS
3       2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291  603123.SS
4       2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935  603123.SS
...            ...    ...    ...    ...    ...        ...       ...        ...
399969  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021  600108.SS
399970  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100  600108.SS
399971  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251  600108.SS
399972  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300  600108.SS
399973  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420  600108.SS

[361606 rows x 8 columns]

In [4]:
df.set_index('code', inplace=True)
df

Date   Open   High    Low  Close  Adj Close    Volume
code                                                                  
603123.SS  2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814
603123.SS  2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265
603123.SS  2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283
603123.SS  2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291
603123.SS  2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935
...               ...    ...    ...    ...    ...        ...       ...
600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021
600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100
600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251
600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300
600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420

[361606 rows x 7 columns]

In [5]:
df['Pct_Chg'] = df.groupby(level=0)['Close'].pct_change()
df['Pct_Chg_5d'] = df.groupby(level=0)['Close'].pct_change(periods=5)
df = df.dropna(subset=['Pct_Chg_5d'])
df

Date   Open   High    Low  Close  Adj Close    Volume  \
code                                                                     
603123.SS  2023-01-10  14.31  14.31  13.61  13.74      13.74  46469718   
603123.SS  2023-01-11  13.87  14.25  13.44  13.67      13.67  40282557   
603123.SS  2023-01-12  13.40  13.64  12.72  12.77      12.77  49225668   
603123.SS  2023-01-13  12.70  12.89  12.64  12.75      12.75  26223929   
603123.SS  2023-01-16  12.85  12.96  12.75  12.85      12.85  22630720   
...               ...    ...    ...    ...    ...        ...       ...   
600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021   
600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100   
600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251   
600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300   
600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420   

            Pct_Chg  Pct_Chg_5d  
code                             
603123.SS -0.039161   -0.084000  
603123.SS -0.005095   -0.051353  
603123.SS -0.065838   -0.152621  
603123.SS -0.001566   -0.102113  
603123.SS  0.007843   -0.101399  
...             ...         ...  
600108.SS  0.016807   -0.012245  
600108.SS -0.008264   -0.004149  
600108.SS  0.008333    0.016807  
600108.SS -0.004132    0.025532  
600108.SS  0.012448    0.025210  

[357096 rows x 9 columns]

In [6]:
stock_data = df.index.get_level_values('code').unique()
stock_data = pd.DataFrame({
    'code': stock_data,
    'is_crash': 0,  # 初始化is_crash列为0
    'is_low': 0
})
stock_data

code  is_crash  is_low
0    603123.SS         0       0
1    600676.SS         0       0
2    600841.SS         0       0
3    601162.SS         0       0
4    600148.SS         0       0
..         ...       ...     ...
896  600370.SS         0       0
897  600202.SS         0       0
898  600985.SS         0       0
899  603188.SS         0       0
900  600108.SS         0       0

[901 rows x 3 columns]

### 这里设置前期暴跌的条件是5日跌20%

In [7]:
df = df.reset_index()
# 遍历df中的每一行
for index, row in df.iterrows():
    # 检查Pct_Chg_5d是否小于-0.2
    if row['Pct_Chg_5d'] < -0.2:
        # 如果是，更新stock_data中的is_crash为1
        stock_data.loc[stock_data['code'] == row['code'], 'is_crash'] = 1
stock_data

code  is_crash  is_low
0    603123.SS         1       0
1    600676.SS         1       0
2    600841.SS         1       0
3    601162.SS         0       0
4    600148.SS         1       0
..         ...       ...     ...
896  600370.SS         1       0
897  600202.SS         1       0
898  600985.SS         0       0
899  603188.SS         1       0
900  600108.SS         0       0

[901 rows x 3 columns]

In [8]:
count_1 = (stock_data['is_crash'] == 1).sum()
count_0 = (stock_data['is_crash'] == 0).sum()
print(f"Number of 1's: {count_1}")
print(f"Number of 0's: {count_0}")

Number of 1's: 398
Number of 0's: 503


### 有398支股票经历过大跌

## 长期慢跌的判断

#### 我认为，如果在过去60天的交易日内，存在60%及以上的数据符合 MA10 < MA20 < MA30 ，则可以认为这只股票处于长期的下跌趋势中

In [9]:
df['MA5'] = df['Close'].rolling(window=5).mean()
df['MA10'] = df['Close'].rolling(window=10).mean()
df['MA20'] = df['Close'].rolling(window=20).mean()
df['MA30'] = df['Close'].rolling(window=30).mean()
df = df.dropna(subset=['MA5','MA10','MA20','MA30'])
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000  
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333  
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333  
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333  
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667  
...          ...       ...         ...     ...     ...      ...        ...  
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000  
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667  
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333  
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000  
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667  

[357067 rows x 14 columns]

In [12]:
df['Condition'] = (df['MA10'] < df['MA20']) & (df['MA20'] < df['MA30'])
df['Condition'] = df['Condition'].astype(int)
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition  
29              1  
30              0  
31              0  
32              1  
33              1  
...           ...  
357091          0  
357092          0  
357093          0  
357094          0  
357095          0  

[357067 rows x 15 columns]

In [13]:
df['low'] = df['Condition'].rolling(window=60).mean()
df_low = df[df['low'] > 0.6]
df_low

code        Date   Open   High    Low  Close  Adj Close  \
88      603123.SS  2023-05-25   9.99   9.99   9.66   9.80   9.800000   
89      603123.SS  2023-05-26   9.80   9.87   9.68   9.84   9.840000   
90      603123.SS  2023-05-29   9.86   9.92   9.73   9.74   9.740000   
91      603123.SS  2023-05-30   9.87  10.45   9.69  10.21  10.210000   
92      603123.SS  2023-05-31  10.10  10.28  10.07  10.13  10.130000   
...           ...         ...    ...    ...    ...    ...        ...   
356986  600108.SS  2024-03-22   2.80   2.80   2.73   2.74   2.733849   
356987  600108.SS  2024-03-25   2.75   2.77   2.73   2.73   2.723871   
356988  600108.SS  2024-03-26   2.73   2.75   2.70   2.73   2.723871   
356989  600108.SS  2024-03-27   2.72   2.73   2.70   2.70   2.693939   
356990  600108.SS  2024-03-28   2.70   2.72   2.68   2.69   2.683961   

          Volume   Pct_Chg  Pct_Chg_5d    MA5    MA10     MA20       MA30  \
88      11638600 -0.023904   -0.032576  9.988  10.037  10.4335  10.617000   
89       6185601  0.004082   -0.017964  9.952  10.007  10.3660  10.562000   
90       6095627 -0.010163   -0.026000  9.900   9.964  10.2800  10.498667   
91      18295788  0.048255    0.012897  9.926   9.987  10.2425  10.448333   
92      10475242 -0.007835    0.008964  9.944   9.999  10.1865  10.401000   
...          ...       ...         ...    ...     ...      ...        ...   
356986  17512777 -0.021429   -0.038596  2.792   2.764   2.7270   2.665667   
356987  14431000 -0.003650   -0.031915  2.774   2.769   2.7285   2.674000   
356988  14467681  0.000000   -0.025000  2.760   2.771   2.7295   2.688333   
356989  10621400 -0.010989   -0.035714  2.740   2.769   2.7295   2.696333   
356990  14058500 -0.003704   -0.039286  2.718   2.766   2.7275   2.705667   

        Condition       low  
88              1  0.866667  
89              1  0.866667  
90              1  0.883333  
91              1  0.900000  
92              1  0.900000  
...           ...       ...  
356986          0  0.683333  
356987          0  0.666667  
356988          0  0.650000  
356989          0  0.633333  
356990          0  0.616667  

[72651 rows x 16 columns]

In [14]:
df_low = df_low.reset_index()
# 遍历df中的每一行
for index, row in df_low.iterrows():
    stock_data.loc[stock_data['code'] == row['code'], 'is_low'] = 1
stock_data

code  is_crash  is_low
0    603123.SS         1       1
1    600676.SS         1       1
2    600841.SS         1       1
3    601162.SS         0       1
4    600148.SS         1       1
..         ...       ...     ...
896  600370.SS         1       1
897  600202.SS         1       1
898  600985.SS         0       1
899  603188.SS         1       1
900  600108.SS         0       1

[901 rows x 3 columns]

In [15]:
count_1_low = (stock_data['is_low'] == 1).sum()
count_0_low = (stock_data['is_low'] == 0).sum()
print(f"Number of 1's: {count_1_low}")
print(f"Number of 0's: {count_0_low}")

Number of 1's: 868
Number of 0's: 33


In [16]:
stock_data_condition = stock_data[(stock_data['is_crash'] == 1) & (stock_data['is_low'] == 1)]
stock_data_condition

code  is_crash  is_low
0    603123.SS         1       1
1    600676.SS         1       1
2    600841.SS         1       1
4    600148.SS         1       1
5    600650.SS         1       1
..         ...       ...     ...
894  605018.SS         1       1
895  600816.SS         1       1
896  600370.SS         1       1
897  600202.SS         1       1
899  603188.SS         1       1

[389 rows x 3 columns]

#### 一共有108支股票符合前期暴跌并且长期慢跌的条件

## 计算最低价格

### 最低价格我认为是在前期经历下跌后在低位的涨停前10天左右的最低价格

那么如何确定第一个涨停板呢，在涨停板出现前大概率是下跌趋势

In [17]:
df_copy = pd.read_csv('./data/all_data_SS_2y.csv')
df_copy = df_copy.drop(df_copy.columns[0], axis=1)
df_copy

Date   Open   High    Low  Close  Adj Close    Volume       code
0       2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814  603123.SS
1       2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265  603123.SS
2       2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283  603123.SS
3       2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291  603123.SS
4       2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935  603123.SS
...            ...    ...    ...    ...    ...        ...       ...        ...
399969  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021  600108.SS
399970  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100  600108.SS
399971  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251  600108.SS
399972  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300  600108.SS
399973  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420  600108.SS

[399974 rows x 8 columns]

In [18]:
df_copy = df_copy[df_copy['code'].isin(stock_data_condition['code'])]
df_copy.set_index('code', inplace=True)
df_copy['Pct_Chg'] = df_copy.groupby(level=0)['Close'].pct_change()
df_copy = df_copy.dropna(subset=['Pct_Chg'])
df_copy

D:\anaconda3\envs\strategy\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


Date   Open   High    Low  Close  Adj Close    Volume  \
code                                                                     
603123.SS  2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265   
603123.SS  2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283   
603123.SS  2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291   
603123.SS  2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935   
603123.SS  2023-01-10  14.31  14.31  13.61  13.74      13.74  46469718   
...               ...    ...    ...    ...    ...        ...       ...   
603188.SS  2024-08-26   2.35   2.43   2.35   2.40       2.40   4149700   
603188.SS  2024-08-27   2.42   2.43   2.34   2.35       2.35   3917400   
603188.SS  2024-08-28   2.36   2.45   2.33   2.42       2.42   7598501   
603188.SS  2024-08-29   2.43   2.43   2.34   2.41       2.41   5812800   
603188.SS  2024-08-30   2.43   2.50   2.39   2.46       2.46   7322801   

            Pct_Chg  
code                 
603123.SS -0.039333  
603123.SS  0.045802  
603123.SS -0.057731  
603123.SS  0.007042  
603123.SS -0.039161  
...             ...  
603188.SS  0.016949  
603188.SS -0.020833  
603188.SS  0.029787  
603188.SS -0.004132  
603188.SS  0.020747  

[154931 rows x 8 columns]

In [19]:
min_close_info = df.loc[df.groupby('code')['Close'].idxmin(), ['code', 'Close', 'Date']]
min_close_info.columns = ['code', 'Close_low','Date']
min_close_info

code  Close_low        Date
38971   600000.SS   6.500000  2024-01-12
197836  600004.SS   8.720000  2024-08-22
84059   600006.SS   5.000000  2024-02-05
180734  600007.SS  16.150000  2023-01-10
109979  600008.SS   2.540000  2024-01-22
...           ...        ...         ...
21887   605588.SS  29.209999  2023-01-11
160319  605589.SS  16.100000  2024-02-05
8641    605598.SS  14.450000  2024-02-06
319859  605599.SS   9.640000  2023-02-16
145458  689009.SS  23.340000  2024-01-31

[901 rows x 3 columns]

In [20]:
df_copy = df_copy.merge(min_close_info, on=['code', 'Date'], how='left')
# 添加 is_lowest 列
df_copy['is_lowest'] = df_copy['Close'].eq(df_copy['Close_low']).astype(int)
df_copy

code        Date   Open   High    Low  Close  Adj Close  \
0       603123.SS  2023-01-04  14.89  14.95  14.35  14.41      14.41   
1       603123.SS  2023-01-05  14.70  15.58  14.50  15.07      15.07   
2       603123.SS  2023-01-06  14.69  14.80  14.14  14.20      14.20   
3       603123.SS  2023-01-09  14.11  14.85  13.95  14.30      14.30   
4       603123.SS  2023-01-10  14.31  14.31  13.61  13.74      13.74   
...           ...         ...    ...    ...    ...    ...        ...   
154926  603188.SS  2024-08-26   2.35   2.43   2.35   2.40       2.40   
154927  603188.SS  2024-08-27   2.42   2.43   2.34   2.35       2.35   
154928  603188.SS  2024-08-28   2.36   2.45   2.33   2.42       2.42   
154929  603188.SS  2024-08-29   2.43   2.43   2.34   2.41       2.41   
154930  603188.SS  2024-08-30   2.43   2.50   2.39   2.46       2.46   

          Volume   Pct_Chg  Close_low  is_lowest  
0       52774265 -0.039333        NaN          0  
1       88029283  0.045802        NaN          0  
2       58946291 -0.057731        NaN          0  
3       43870935  0.007042        NaN          0  
4       46469718 -0.039161        NaN          0  
...          ...       ...        ...        ...  
154926   4149700  0.016949        NaN          0  
154927   3917400 -0.020833        NaN          0  
154928   7598501  0.029787        NaN          0  
154929   5812800 -0.004132        NaN          0  
154930   7322801  0.020747        NaN          0  

[154931 rows x 11 columns]

写到这里我感觉由最低价得出第一个涨停板过于困难，于是我打算尝试先找涨停板

那么如何区分低位涨停板和高位涨停板呢？

我认为在均线 MA5 < MA10 < MA20 < MA30出现的涨停板可认为是最低点的涨停板

In [21]:
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  
29              1       NaN  
30              0       NaN  
31              0       NaN  
32              1       NaN  
33              1       NaN  
...           ...       ...  
357091          0  0.366667  
357092          0  0.366667  
357093          0  0.366667  
357094          0  0.366667  
357095          0  0.350000  

[357067 rows x 16 columns]

In [22]:
df['满足条件'] = ((df['Condition'] == 1) & (df['Pct_Chg'] > 0.098) & (df['low'] > 0.5)).astype(int)
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  满足条件  
29              1       NaN     0  
30              0       NaN     0  
31              0       NaN     0  
32              1       NaN     0  
33              1       NaN     0  
...           ...       ...   ...  
357091          0  0.366667     0  
357092          0  0.366667     0  
357093          0  0.366667     0  
357094          0  0.366667     0  
357095          0  0.350000     0  

[357067 rows x 17 columns]

In [23]:
# condition_met_indices = df[df['满足条件'] == 1].index
# df['Date'] = pd.to_datetime(df['Date'])
# for index in condition_met_indices:
#     # 获取前十天的范围
#     start_date = df.loc[index, 'Date']
#     end_date = start_date - pd.Timedelta(days=10)
    
#     # 选取前十天的数据
#     mask = (df['Date'] >= end_date) & (df['Date'] <= start_date)
#     previous_days_df = df.loc[mask]
    
#     # 计算最低点
#     lowest_price = previous_days_df['Close'].min()
    
#     # 将最低点添加到新列'price_lowest'中
#     df.loc[index, 'price_lowest'] = lowest_price
# df

In [24]:
# df_test = df
# condition_met_indices = df_test[df_test['满足条件'] == 1].index
# df_test['Date'] = pd.to_datetime(df_test['Date'])
# for index in condition_met_indices:
#     # 获取前十天的范围
#     start_date = df_test.loc[index, 'Date']
#     end_date = start_date - pd.Timedelta(days=10)
    
#     # 选取前十天的数据
#     mask = (df_test['Date'] >= end_date) & (df_test['Date'] <= start_date)
#     previous_days_df = df_test.loc[mask]
    
#     # 计算最低点
#     lowest_price = previous_days_df['Close'].min()
    
#     # 将最低点添加到新列'price_lowest'中
#     df_test.loc[index, 'price_lowest'] = lowest_price
# pd.set_option('display.max_rows', None)  # 设置显示所有行
# df_test

In [25]:
df_test = df
df_test

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  满足条件  
29              1       NaN     0  
30              0       NaN     0  
31              0       NaN     0  
32              1       NaN     0  
33              1       NaN     0  
...           ...       ...   ...  
357091          0  0.366667     0  
357092          0  0.366667     0  
357093          0  0.366667     0  
357094          0  0.366667     0  
357095          0  0.350000     0  

[357067 rows x 17 columns]

### 最低价为低位涨停板前十天最低价

In [26]:
df_test['最低价'] = df_test.groupby('code')['Close'].transform(lambda x: x.rolling(window=10, min_periods=1).min())
df_test

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  满足条件    最低价  
29              1       NaN     0  12.09  
30              0       NaN     0  12.09  
31              0       NaN     0  12.09  
32              1       NaN     0  12.09  
33              1       NaN     0  12.09  
...           ...       ...   ...    ...  
357091          0  0.366667     0   2.35  
357092          0  0.366667     0   2.35  
357093          0  0.366667     0   2.35  
357094          0  0.366667     0   2.35  
357095          0  0.350000     0   2.35  

[357067 rows x 18 columns]

In [27]:
df_test['出货价'] = df_test['最低价'] * 1.86
df_test

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  满足条件    最低价      出货价  
29              1       NaN     0  12.09  22.4874  
30              0       NaN     0  12.09  22.4874  
31              0       NaN     0  12.09  22.4874  
32              1       NaN     0  12.09  22.4874  
33              1       NaN     0  12.09  22.4874  
...           ...       ...   ...    ...      ...  
357091          0  0.366667     0   2.35   4.3710  
357092          0  0.366667     0   2.35   4.3710  
357093          0  0.366667     0   2.35   4.3710  
357094          0  0.366667     0   2.35   4.3710  
357095          0  0.350000     0   2.35   4.3710  

[357067 rows x 19 columns]

### 最高价为第一个涨停板后30天内的最高价

In [28]:
df_test['最高价'] = df_test.groupby('code')['Close'].transform(lambda x: x.shift(-1).rolling(window=30, min_periods=1).max())
df_test

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  满足条件    最低价      出货价    最高价  
29              1       NaN     0  12.09  22.4874  12.25  
30              0       NaN     0  12.09  22.4874  12.35  
31              0       NaN     0  12.09  22.4874  12.35  
32              1       NaN     0  12.09  22.4874  12.35  
33              1       NaN     0  12.09  22.4874  12.38  
...           ...       ...   ...    ...      ...    ...  
357091          0  0.366667     0   2.35   4.3710   2.49  
357092          0  0.366667     0   2.35   4.3710   2.49  
357093          0  0.366667     0   2.35   4.3710   2.49  
357094          0  0.366667     0   2.35   4.3710   2.49  
357095          0  0.350000     0   2.35   4.3710   2.49  

[357067 rows x 20 columns]

In [33]:
df_match = df_test[df_test['满足条件'] == 1 ]
# pd.set_option('display.max_rows', None)
df_match

code        Date       Open       High        Low      Close  \
264     603123.SS  2024-02-08   6.740000   7.470000   6.730000   7.470000   
1160    600841.SS  2024-07-11   3.280000   3.530000   3.230000   3.530000   
1161    600841.SS  2024-07-12   3.500000   3.880000   3.450000   3.880000   
1863    600148.SS  2024-02-21  11.280000  12.470000  11.040000  12.470000   
3161    605218.SS  2024-07-19  14.710000  16.200001  14.650000  16.200001   
3208    600847.SS  2023-02-08  14.800000  16.299999  14.710000  16.299999   
3989    603883.SS  2024-08-30  12.260000  13.500000  11.920000  13.500000   
4795    600611.SS  2023-01-19   3.280000   3.280000   3.280000   3.280000   
5549    603626.SS  2024-07-11   5.100000   5.500000   5.090000   5.500000   
5839    600826.SS  2024-01-24   5.842857   6.371428   5.842857   6.371428   
5850    600826.SS  2024-02-08   5.214285   5.357142   5.142857   5.357142   
5851    600826.SS  2024-02-19   5.892857   5.892857   5.892857   5.892857   
6647    600418.SS  2024-02-07  11.380000  12.560000  11.380000  12.560000   
8838    600769.SS  2023-04-12   7.440000   8.160000   7.440000   8.160000   
9556    600576.SS  2024-08-05   4.110000   4.400000   4.030000   4.400000   
9557    600576.SS  2024-08-06   4.830000   4.840000   4.460000   4.840000   
9976    600898.SS  2023-01-17   5.640000   6.200000   5.610000   6.200000   
11423   605228.SS  2024-02-08   6.770000   7.270000   6.180000   7.270000   
11424   605228.SS  2024-02-19   7.290000   8.000000   7.290000   8.000000   
13015   603161.SS  2024-02-08   7.478873   8.260563   7.471830   8.260563   
15367   603378.SS  2024-02-08   5.100000   5.450000   4.750000   5.450000   
15370   603378.SS  2024-02-21   5.770000   6.400000   5.710000   6.400000   
16295   600506.SS  2024-08-26  10.470000  11.520000  10.420000  11.520000   
16484   603388.SS  2023-10-18   7.720000   7.810000   7.110000   7.810000   
16487   603388.SS  2023-10-23   8.080000   8.340000   8.080000   8.340000   
16567   603388.SS  2024-02-21   4.680000   5.250000   4.650000   5.250000   
16963   600326.SS  2024-02-08   2.750000   3.030000   2.740000   3.030000   
17350   600119.SS  2024-01-23   6.730000   6.730000   6.730000   6.730000   
17351   600119.SS  2024-01-24   7.400000   7.400000   7.400000   7.400000   
17352   600119.SS  2024-01-25   8.140000   8.140000   8.140000   8.140000   
18263   600243.SS  2024-07-18   2.590000   2.850000   2.530000   2.850000   
18264   600243.SS  2024-07-19   2.900000   3.140000   2.820000   3.140000   
18847   603918.SS  2023-08-28  27.990000  28.160000  25.969999  28.160000   
18896   603918.SS  2023-11-13  20.520000  22.480000  20.520000  22.480000   
18948   603918.SS  2024-01-25  16.590000  18.580000  16.559999  18.580000   
18959   603918.SS  2024-02-19  13.000000  13.810000  12.990000  13.810000   
19049   603918.SS  2024-07-02  10.500000  11.540000  10.500000  11.540000   
20083   603338.SS  2023-10-30  50.619999  50.619999  50.020000  50.619999   
20544   600171.SS  2024-01-25  11.880000  13.160000  11.800000  13.160000   
20545   600171.SS  2024-01-26  13.560000  14.480000  13.300000  14.480000   
21098   601028.SS  2023-01-31  11.990000  13.180000  11.810000  13.180000   
21352   601028.SS  2024-02-08   7.880000   8.500000   7.880000   8.500000   
23793   601995.SS  2024-04-26  32.009998  33.910000  31.490000  33.910000   
24623   603268.SS  2024-06-17  11.850000  12.770000  11.450000  12.770000   
24624   603268.SS  2024-06-18  12.900000  14.050000  12.890000  14.050000   
26139   603822.SS  2024-02-08  15.920000  17.639999  15.920000  17.639999   
26140   603822.SS  2024-02-19  19.400000  19.400000  18.650000  19.400000   
26475   601133.SS  2024-02-08   8.580000   9.330000   8.490000   9.330000   
26484   601133.SS  2024-02-29  10.160000  11.030000   9.950000  11.030000   
27959   603931.SS  2023-08-31  27.309999  29.870001  26.900000  29.870001   
28067   603931.SS  2024-02-08  18.230000  19.930000  18.030001  19.930000   
28072   6

In [34]:
df_match_1 = df_match[df_match['最高价'] >= df_match['出货价']]
df_match_1

code        Date  Open   High   Low  Close  Adj Close     Volume  \
16567   603388.SS  2024-02-21  4.68   5.25  4.65   5.25   5.250000   38086980   
32057   603363.SS  2024-02-08  2.38   2.90  2.38   2.90   2.900000   85989369   
32059   603363.SS  2024-02-20  2.83   3.15  2.78   3.15   3.150000   43210984   
32061   603363.SS  2024-02-22  3.15   3.54  3.15   3.54   3.540000   60596365   
32062   603363.SS  2024-02-23  3.65   3.89  3.57   3.89   3.890000   65922575   
32063   603363.SS  2024-02-26  4.11   4.28  3.99   4.28   4.280000   66490609   
32064   603363.SS  2024-02-27  4.35   4.71  4.20   4.71   4.710000   53361155   
37885   600811.SS  2024-07-03  0.76   0.77  0.74   0.77   0.770000   82004411   
37886   600811.SS  2024-07-04  0.84   0.85  0.77   0.85   0.850000  386747182   
37891   600811.SS  2024-07-11  0.85   0.88  0.83   0.88   0.880000  236653382   
37892   600811.SS  2024-07-12  0.91   0.97  0.90   0.97   0.970000  395133365   
46568   600375.SS  2024-02-27  5.35   5.89  5.18   5.89   5.890000   36526967   
49360   600889.SS  2024-02-26  4.09   4.49  4.09   4.49   4.490000    9165802   
51755   603879.SS  2024-02-27  4.50   4.93  4.49   4.93   4.930000   11420244   
79275   605055.SS  2024-02-19  4.01   4.37  4.01   4.37   4.315714    9477573   
164312  603278.SS  2024-02-08  6.76   7.39  6.05   7.38   7.330039   22013571   
275282  600165.SS  2024-02-21  1.71   1.88  1.68   1.88   1.880000   40413838   
275284  600165.SS  2024-02-23  2.00   2.19  1.96   2.19   2.190000   72584761   
343399  600203.SS  2024-02-08  3.65   4.07  3.57   4.07   4.070000   40151486   
346189  603528.SS  2024-02-19  4.84   5.23  4.84   5.23   5.230000   20718825   
346192  603528.SS  2024-02-22  5.73   6.22  5.68   6.22   6.220000   49148978   
348986  600355.SS  2024-02-23  3.73   3.99  3.70   3.99   3.990000   77043635   
348987  600355.SS  2024-02-26  4.39   4.39  4.28   4.39   4.390000   25108920   
350976  600137.SS  2024-02-08  9.92  10.92  9.66  10.92  10.755652    4347855   

         Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  Condition  \
16567   0.100629    0.105263   4.682   5.144   6.2310   6.859333          1   
32057   0.098485   -0.278607   3.070   3.924   4.6500   5.049000          1   
32059   0.101399   -0.033742   2.896   3.482   4.3770   4.845667          1   
32061   0.099379    0.340909   3.134   3.214   4.1585   4.669667          1   
32062   0.098870    0.341379   3.332   3.201   4.0825   4.603000          1   
32063   0.100257    0.496504   3.616   3.267   4.0285   4.553000          1   
32064   0.100467    0.495238   3.928   3.412   4.0135   4.523000          1   
37885   0.100000   -0.049383   0.700   0.813   1.0800   1.240000          1   
37886   0.103896    0.164384   0.724   0.787   1.0510   1.212000          1   
37891   0.100000    0.035294   0.860   0.792   0.9275   1.089333          1   
37892   0.102273    0.141176   0.884   0.823   0.9080   1.070667          1   
46568   0.100935    0.613699   4.908   4.074   4.4345   4.940667          1   
49360   0.100490    0.275568   3.960   3.664   4.2895   4.751000          1   
51755   0.100446    0.339674   4.382   3.959   4.5830   5.105667          1   
79275   0.100756   -0.122490   4.104   4.878   5.7995   6.277667          1   
164312  0.098214   -0.157534   7.564   8.461   9.5410  10.339667          1   
275282  0.099415    0.074286   1.696   1.917   2.4065   2.697333          1   
275284  0.100503    0.303572   1.880   1.850   2.3105   2.610000          1   
343399  0.100000   -0.227704   4.260   5.043   5.7060   6.135333          1   
346189  0.101053   -0.132670   4.946   5.827   6.9160   7.522333          1   
346192  0.100885    0.407240   5.476   5.461   6.5160   7.204667          1   
348986  0.099174    0.608871   3.330   3.029   3.4640   3.771667          1   
348987  0.100251    0.608059   3.662   3.144   3.4725   3.768667          1   
350976  0.099698   -0.214388  11.198  13.184  15.0620  16.096667          1   

           

### 计算主力出货平均涨幅

In [35]:
df_match['主力出货涨幅'] = df_match['最高价'] / df_match['最低价'] - 1
df_match

D:\anaconda3\envs\strategy\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


code        Date       Open       High        Low      Close  \
264     603123.SS  2024-02-08   6.740000   7.470000   6.730000   7.470000   
1160    600841.SS  2024-07-11   3.280000   3.530000   3.230000   3.530000   
1161    600841.SS  2024-07-12   3.500000   3.880000   3.450000   3.880000   
1863    600148.SS  2024-02-21  11.280000  12.470000  11.040000  12.470000   
3161    605218.SS  2024-07-19  14.710000  16.200001  14.650000  16.200001   
3208    600847.SS  2023-02-08  14.800000  16.299999  14.710000  16.299999   
3989    603883.SS  2024-08-30  12.260000  13.500000  11.920000  13.500000   
4795    600611.SS  2023-01-19   3.280000   3.280000   3.280000   3.280000   
5549    603626.SS  2024-07-11   5.100000   5.500000   5.090000   5.500000   
5839    600826.SS  2024-01-24   5.842857   6.371428   5.842857   6.371428   
5850    600826.SS  2024-02-08   5.214285   5.357142   5.142857   5.357142   
5851    600826.SS  2024-02-19   5.892857   5.892857   5.892857   5.892857   
6647    600418.SS  2024-02-07  11.380000  12.560000  11.380000  12.560000   
8838    600769.SS  2023-04-12   7.440000   8.160000   7.440000   8.160000   
9556    600576.SS  2024-08-05   4.110000   4.400000   4.030000   4.400000   
9557    600576.SS  2024-08-06   4.830000   4.840000   4.460000   4.840000   
9976    600898.SS  2023-01-17   5.640000   6.200000   5.610000   6.200000   
11423   605228.SS  2024-02-08   6.770000   7.270000   6.180000   7.270000   
11424   605228.SS  2024-02-19   7.290000   8.000000   7.290000   8.000000   
13015   603161.SS  2024-02-08   7.478873   8.260563   7.471830   8.260563   
15367   603378.SS  2024-02-08   5.100000   5.450000   4.750000   5.450000   
15370   603378.SS  2024-02-21   5.770000   6.400000   5.710000   6.400000   
16295   600506.SS  2024-08-26  10.470000  11.520000  10.420000  11.520000   
16484   603388.SS  2023-10-18   7.720000   7.810000   7.110000   7.810000   
16487   603388.SS  2023-10-23   8.080000   8.340000   8.080000   8.340000   
16567   603388.SS  2024-02-21   4.680000   5.250000   4.650000   5.250000   
16963   600326.SS  2024-02-08   2.750000   3.030000   2.740000   3.030000   
17350   600119.SS  2024-01-23   6.730000   6.730000   6.730000   6.730000   
17351   600119.SS  2024-01-24   7.400000   7.400000   7.400000   7.400000   
17352   600119.SS  2024-01-25   8.140000   8.140000   8.140000   8.140000   
18263   600243.SS  2024-07-18   2.590000   2.850000   2.530000   2.850000   
18264   600243.SS  2024-07-19   2.900000   3.140000   2.820000   3.140000   
18847   603918.SS  2023-08-28  27.990000  28.160000  25.969999  28.160000   
18896   603918.SS  2023-11-13  20.520000  22.480000  20.520000  22.480000   
18948   603918.SS  2024-01-25  16.590000  18.580000  16.559999  18.580000   
18959   603918.SS  2024-02-19  13.000000  13.810000  12.990000  13.810000   
19049   603918.SS  2024-07-02  10.500000  11.540000  10.500000  11.540000   
20083   603338.SS  2023-10-30  50.619999  50.619999  50.020000  50.619999   
20544   600171.SS  2024-01-25  11.880000  13.160000  11.800000  13.160000   
20545   600171.SS  2024-01-26  13.560000  14.480000  13.300000  14.480000   
21098   601028.SS  2023-01-31  11.990000  13.180000  11.810000  13.180000   
21352   601028.SS  2024-02-08   7.880000   8.500000   7.880000   8.500000   
23793   601995.SS  2024-04-26  32.009998  33.910000  31.490000  33.910000   
24623   603268.SS  2024-06-17  11.850000  12.770000  11.450000  12.770000   
24624   603268.SS  2024-06-18  12.900000  14.050000  12.890000  14.050000   
26139   603822.SS  2024-02-08  15.920000  17.639999  15.920000  17.639999   
26140   603822.SS  2024-02-19  19.400000  19.400000  18.650000  19.400000   
26475   601133.SS  2024-02-08   8.580000   9.330000   8.490000   9.330000   
26484   601133.SS  2024-02-29  10.160000  11.030000   9.950000  11.030000   
27959   603931.SS  2023-08-31  27.309999  29.870001  26.900000  29.870001   
28067   603931.SS  2024-02-08  18.230000  19.930000  18.030001  19.930000   
28072   6

In [37]:
average_gain = df_match['主力出货涨幅'].mean()
average_gain

0.3760138544219926

## 这里得到主力出货的平均涨幅在0.37